### ISSUE TITLE
[Assert] UI Canvases are not drawn when entering the Game Mode

### ISSUE THREAD

**Issue by LB-ArturSiewniak** [o3de/o3de#5403]
**Describe the bug**
Entering the Game Mode with visible .uicanvas asset assigned to the UI Canvas Asset Ref component causes an assert to pop up, and the assigned .uicanvas asset is not being displayed.

This issue does not occur on the Development (3b1c633) build.

This issue can be related to https://github.com/o3de/o3de/issues/5402.

File:
```
C:/Builds/o3de/Gems/Atom/RPI/Code/Source/RPI.Public/DynamicDraw/DynamicDrawContext.cpp
```
Condition:
```
(3012) 'void __cdecl AZ::RPI::DynamicDrawContext::DrawIndexed(const void *,unsigned int,const void *,unsigned int,enum AZ::RHI::IndexFormat,class AZStd::intrusive_ptr<class AZ::RPI::ShaderResourceGroup>)'
Can't add draw calls after draw data was finalized
```
Line: 456

**Steps to reproduce**
1. Execute the sys_asserts 2 command in the Console.
2. Create an entity and add a UI Canvas Asset Ref component to it.
3. Enable the Load automatically parameter.
4. Click the Open in UI Editor button next to the Canvas pathname parameter.
5. Create a simple visible .uicanvas asset (e.g. Create a new empty element and add an Image component to it) and save it. 
6. Make sure that the saved .uicanvas asset from step 5 is assigned in the UI Canvas Asset Ref component.
7. Enter the Game Mode.

**Expected behavior**
UI Canvas is displayed.

**Actual behavior**
UI Canvas is not being displayed, an assert pops up.

**Video, assert, and Editor.log**

https://user-images.githubusercontent.com/86952082/140752032-a4d6a21f-4b40-4898-a994-23e165e07e4c.mp4

Assert:
![AssertUI](https://user-images.githubusercontent.com/86952082/140752159-b29698ba-ef13-4cd0-9ccb-f71447bb88bb.PNG)

[Editor.log](https://github.com/o3de/o3de/files/7497353/Editor.log)

**Found in Branch**
Stabilization_2110 (621194e)

**Desktop**
Device: PC
OS: Windows
Version 10
CPU AMD Ryzen 5 3600 6-Core Processor, 3600 MHz, 6 Core(s), 12 Logical Processor(s)
GPU NVIDIA GeForce RTX 2060 SUPER
Memory 16GB

**Comment by chiyenteng** [comment]
@VickyAtAZ have fixed that issue in development and she just cheery-picked this fix to stable an hour ago: https://github.com/o3de/o3de/pull/5389

**Comment by LB-ArturSiewniak** [comment]
Closing the issue. Verified as fixed on build Stabilization_2110 (76b7e73).
